### DISCUSSION TITLE
The combo color 1 is not the first combo color appeared in the map

### DISCUSSION THREAD

**Discussion by cihe13375** [ppy/osu#27316] [Q&A] [answered]
Combo color settings:
![11](https://github.com/ppy/osu/assets/33959010/03053c6a-8b3d-421c-92a6-d44874538c48)
At the beginning of the map (note that the "Combo 2" appears first):
![12](https://github.com/ppy/osu/assets/33959010/55ea490a-e310-4195-a00f-dca59b96de85)
When exporting to stable, the "Combo 2" (in lazer editor) becomes Combo 1 (in stable song setup).

**Comment by bdach** [comment]
Please link map.

**Reply by cihe13375** [reply]
This occurs on any map.

**Reply by bdach** [reply]
I want to make sure there is no weird "color hax" involved or whatever. And work with a concrete example rather than guessing.

**Reply by cihe13375** [reply]
How I created the example above:
1) open lazer editor from the main menu
2) set combo colors in the setup panel as shown in the 1st picture
3) add a timing point (mostly for convenience)
4) place circles in the compose panel as shown in the 2nd picture

or if you prefer an existing map (I'm just linking a random map in my installation): https://osu.ppy.sh/beatmapsets/29107#osu/97397 (note that the green color is combo 2 and appears first)

**Comment by OwenCMYK** [comment]
Probably not useful, but I can confirm this is true for me as well on every beatmap I've checked. Which as of right now is:
- https://osu.ppy.sh/beatmapsets/29107#osu/97397 (the one cihe linked)
- https://osu.ppy.sh/beatmapsets/966568#osu/2048440
- https://osu.ppy.sh/beatmapsets/1173192#osu/2446833
- https://osu.ppy.sh/beatmapsets/33842#osu/110175
They all start on color 2, so I'm pretty sure this is a universal thing

**Comment by bdach** [comment]
Well the issue here is that stable basically lies.

What's in the `.osu` file:

```
[Colours]
Combo1 : 0,0,255
Combo2 : 0,255,0
Combo3 : 255,128,0
Combo4 : 255,0,0
Combo5 : 244,0,244
```

In order, these are: `#0000ff` `#00ff00` `#ff8000` `#ff0000` `#f400f4`

What's shown in the actual game:

![image](https://github.com/ppy/osu/assets/20418176/96148dc4-06b9-4c94-945f-defa89c52d99)

There is actual stable code that swaps puts the first colour at the end instead. lazer's only fault here is taking what's written to the `.osu` file at face value.

Unsure of the least bad way of fixing it code-quality wise.

**Reply by OwenCMYK** [reply]
I think this feels like it should be fixed, so maybe lie like stable on the older beatmaps, but for beatmaps made in lazer do it as the file says? Or potentially just continue to use the old stable way forever more. I think those are the only two real possibilities because I feel like re-ordering the colors for any beatmaps is a no-go because there are a lot of maps that use those to represent the song in unique ways.

**Comment by bdach** [comment, marked as answer]
https://github.com/ppy/osu/pull/27344 maybe
